### PR TITLE
Rewrite tests to allow for different DBMSs to be tested

### PR DESCRIPTION
**Pull request by markdirish**
Previously, the tests were written exclusively for IBM i. Now, a user must pass in a `DBMS` environment variable when running the tests, which will load a particular `.env` file and `DBMS/${global.dbms}/config.js` file to run the tests with the given DBMS. This allows the tests to be checked against different drivers and databases, which will help catch many more bugs before publishing the package.

(One bug caught by MSSQL driver was that setting a statement attribute of `SQL_ATTR_ROW_ARRAY_SIZE` can (and probably should) be done right before fetching, not before prepare/bind/execute workflow).

### CODE REVIEW

File: test/connection/beginTransaction.js [hunk 1]
```diff
@@ -104,36 +94,44 @@ describe('.beginTransaction([callback])...', () => {
...
+      if (global.dbms === 'ibmi')
+      {
+        // set commitment level to 1 (Read committed)
+        connectionString1Keywords= ';CMT=1';
+        // set commitment level to 1 (Read committed)
+        connectionString2Keywords = ';CMT=1;CONCURRENTACCESSRESOLUTION=1';
+      }
+      else if (global.dbms === 'mssql')
+      {
+        connectionString2Keywords = ';DMConnAttr=[1227]=32';
```
Comment (kadler) [inline test/connection/beginTransaction.js:108]: FYI, this is a unixODBC-specific attribute

File: test/statement/close.js [hunk 2]
```diff
@@ -124,7 +122,12 @@ describe('.close([calback])...', () => {
         });
       });
     });
-    it('...should close after calling prepare with an error (bad sql prepared).', (done) => {
+    it('...should close after calling prepare with an error (bad sql prepared).', function(done) {
+      // SQL Server doesn't check for syntax error when the application calls SQLPrepare
+      if (global.dbms === 'mssql')
+      {
+        return this.skip('aix!');
```
Comment (kadler) [inline test/statement/close.js:129]: aix?

File: test/statement/close.js [hunk 3]
```diff
@@ -243,7 +246,12 @@ describe('.close([calback])...', () => {
       assert.notDeepEqual(result, null);
       await statement.close();
     });
-    it('...should close after calling prepare with an error (bad sql prepared).', async () => {
+    it('...should close after calling prepare with an error (bad sql prepared).', async function() {
+      // SQL Server doesn't check for syntax error when the application calls SQLPrepare
+      if (global.dbms === 'mssql')
+      {
+        return this.skip('aix!');
```
Comment (kadler) [inline test/statement/close.js:253]: ditto

File: test/connection/beginTransaction.js (hunk 1 above)
Comment (markdirish) [reply test/connection/beginTransaction.js:108]: Definitely something that I will have to tackle in the future... Not sure what else would be a good way of setting these driver-specific values without the driver header file (not that even that would help in these test cases). Maybe a generic "setConnectionAttribute" function where I pass in the magic numbers?

File: test/statement/close.js (hunk 2 above)
Comment (markdirish) [reply test/statement/close.js:129]: Ahh yeah, I was trying to see if the skip function would print when skipped (it doesn't), and I think it was during our meeting and the last words I heard were "aix".

File: test/connection/beginTransaction.js (hunk 1 above)
Comment (kadler) [reply test/connection/beginTransaction.js:108]: Yeah, I think going this way is fine for now. Might be worth adding a Windows check that exits with an explanatory message, though.

A generic interface would be good. Can always add niceties on top of it if needed. I don't think it's necessary if all it's needed for at the moment is testing.

**Review by kadler — APPROVED** [review]
LGTM

**Pull request merged**